src/PhpSpreadsheet/Reader/Gnumeric.php

```diff
@@ -77,16 +82,18 @@ public function canRead($pFilename)
         if (function_exists('gzread')) {
             // Read signature data (first 3 bytes)
             $fh = fopen($pFilename, 'rb');
-            $data = fread($fh, 2);
-            fclose($fh);
+            if ($fh !== false) {
+                $data = fread($fh, 2);
+                fclose($fh);
+            }
         }
 
         return $data == chr(0x1F) . chr(0x8B);
     }
 
     private static function matchXml(string $name, string $field): bool
     {
-        return 1 === preg_match("/^(gnm|gmr):$field$/", $name);
+        return 1 === preg_match("/:$field$/", $name);
```

Comment:
I think the correct test here is against `namespaceUri` (to be sure it's not something with the same name from a different namespace) and `localName` (which is the name of the element without any prefix) instead of against `name` (which might or might not have a prefix, depending on the `xmlns` attributes in scope), e.g.

```php
private static function matchXml(XMLWriter $node, string $expectedLocalName) {
     return
          $xml->namespaceUri == self::NAMESPACE_GNM
          && $xml->localName == $expectedLocalName
          && $xml->nodeType == XMLReader::ELEMENT;
}
```

At a glance, the extra check against nodeType seems to be present everywhere this is used, so could be moved here, e.g. line 145 would become just:

```php
if (self::matchXml($xml, 'Sheet')) {
```

Reply:
Thank you both for the original suggestion, and the new suggestion for how to deal with XMLReader. I have adapted your suggested code and included it in the latest push, and will try to be alert for the attribute problem when I get around to looking at Xlsx.